src/openapi/streaming/connection/connection.ts

```diff
@@ -200,7 +200,16 @@ class Connection {
             return this.createTransport(baseUrl);
         }
 
-        return new SelectedTransport(baseUrl, this.onTransportFail);
+        let options;
+        if (this.options && this.options.isWebsocketStreamingHeartBeatEnabled) {

```

Comment:
there is already a way of passing options - on start - would it be neater to do it in the same way? we don't need the option until it is started..

Reply:
using start method now to set the flag